src/oop/programs_with_multiple_classes/medium/q4/MainTest.java

```diff
@@ -0,0 +1,102 @@
+package oop.programs_with_multiple_classes.medium.q4;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertArrayEquals;
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private final String wheelClass = "Wheel";
+    private final String chassisClass = "Chassis";
+    private final String varDurability = "durability";
+    private final String varManufacturer = "manufacturer";
+    private final String varMaterialType = "materialType";
+    private final String varWheels = "wheels";
+    public ObjectTest wheel;
+    public ObjectTest chassis;
+
+    @BeforeEach
+    public void setup() throws Throwable {
+        String wheelClassString = "oop.programs_with_multiple_classes.medium.q4." + wheelClass;
+        String chassisClassString = "oop.programs_with_multiple_classes.medium.q4." + chassisClass;
+        wheel = new ObjectTest(wheelClassString);
+        chassis = new ObjectTest(chassisClassString);
+        String modifiedWheelMessage =
+                "You have modified the provided portions of the " + wheelClass + " class. Please revert it back to the original state.";
+        assertTrue(wheel.hasField(varDurability, double.class), modifiedWheelMessage);
+        assertTrue(wheel.hasField(varManufacturer, String.class), modifiedWheelMessage);
+        assertTrue(wheel.hasModifier(varDurability, "private"), modifiedWheelMessage);
+        assertTrue(wheel.hasModifier(varManufacturer, "private"), modifiedWheelMessage);
+        String modifiedChassisMessage =
+                "You have modified the provided portions of the " + chassisClass + " class. Please revert it back to the original state.";
+        Class<?>[] classArguments = {String.class};
+        assertTrue(chassis.hasConstructor(classArguments), modifiedChassisMessage);
+        assertTrue(chassis.hasModifier(classArguments, "public"), modifiedChassisMessage);
+        String[] tests = new String[]{"Steel", "Iron", "Carbon", "Titanium"};
+        for (int i = 0; i < tests.length; i++) {
+            Object[][] arguments = {
+                    {tests[i], String.class}
+            };
+            Object chassisInstance = chassis.createInstance(arguments);
+            _assertEquals(tests[i], chassis.getFieldValue(chassisInstance, varMaterialType),
+                    "Your " + chassisClass + " constructor does not correctly initialize the " + varMaterialType + " field.");
```

Comment:
Maybe just say here that they modified the value and it does not initialize the value correctly